### PR TITLE
Move some headers from game.h to .cpp files

### PR DESCRIPTION
**Pull request by Coolthulhu**
`game.h` is included pretty much everywhere, so it should have as little headers as possible.

This should make the game compile a bit faster and make the debug builds smaller.

### CODE REVIEW

**Comment by KA101** [comment]
This failed to build on mine, something about a "timecheck" in an ammo thing.  Sorry I don't have a more precise report, was the first failure I had in testing tonight.

**Comment by Coolthulhu** [comment]
Updated for post-stable.

File: src/game.h [hunk 1]
```diff
@@ -1,28 +1,18 @@
...
-#include "omdata.h"
-#include "crafting.h"
-#include "npc.h"
 #include "faction.h"
 #include "event.h"
 #include "mission.h"
-#include "weather.h"
-#include "construction.h"
 #include "calendar.h"
-#include "posix_time.h"
```
Comment (narc0tiq) [inline src/game.h]: I guess [game.h was using posix_time.h](http://ci.narc.ro/job/Cataclysm-Matrix/Graphics=Curses,Platform=Windows/2855/console), huh?

**Comment by KA101** [comment]
As we discussed on IRC, on mine, that built and ran fine.  Test was both a quick Play Now! and a custom-char Assault on Precinct Z, which thanks to a Thriller had some time for scrounging.

I'll apologize for the build-break nevertheless, but I honestly believed the problem fixed.

**Pull request merged**